### DISCUSSION TITLE
Load attr from subquery

### DISCUSSION THREAD

**Discussion by michaelbukachi** [sqlalchemy/sqlalchemy#8203] [Usage Questions] [answered]
Is it possible to load an "extra" attribute as part of a model?  For instance, if we do a join query with a subquery. :thinking: 
Take  a look at the following snippet
```python
class A(Base):
     id = Column(Integer)
     a = Column(String)

Class B(Base)
     id = Column(Integer)
     b = Column(String)


subquery = select(B.b).subquery("subquery")
query = select(A).outerjoin(subquery, <some condition)
obj = db.execute(query).first()

print(obj.b) # Is it possible to define how b is loaded?
```
Basically, is it possible to define a "load option" that doesn't require a relationship i.e something along the lines of "when loading this model set this attribute from column 'a' of subquery 'subquery'"

**Comment by zzzeek** [comment, marked as answer]
hi there -

no there is not.  you would load the two objects independently and assign them to each other after receiving the result.

usually there's no problem simply putting a viewonly, possibly `lazy=None` relationship on "A" for which you can use [contains_eager](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html?highlight=contains_eager#sqlalchemy.orm.contains_eager) to set it up in any way you'd like
